File: frontend/src/types/dto/entity/game/GamePlayerDto.ts
import type { PlayerColour } from "@/types/enum/PlayerColour";
import type { GameResult } from "@/types/enum/GameResult";
import type { UserDto } from "../user/UserDto";
import type { CellStatus } from "@/types/enum/CellStatus";

export type GamePlayerDto = {
    user: UserDto,
    playerColour: PlayerColour,
    boardProgress: CellStatus[],
    score: number,
    firsts: number,
    mistakes: number,
    maxStreak: number,
    gameLoaded: boolean,
    gameResult: GameResult
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/game/GamePlayerDto.java
package com.github.ryand6.sudokuweb.dto.entity.game;

import com.github.ryand6.sudokuweb.dto.entity.user.UserDto;
import com.github.ryand6.sudokuweb.enums.CellStatus;
import com.github.ryand6.sudokuweb.enums.GameResult;
import com.github.ryand6.sudokuweb.enums.PlayerColour;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GamePlayerDto {

    private UserDto user;

    private PlayerColour playerColour;

    private CellStatus[] boardProgress;

    private int score;

    private int firsts;

    private int mistakes;

    private int maxStreak;

    private boolean gameLoaded;

    private GameResult gameResult;

}
